rmw_implementation/src/functions.cpp

```diff
@@ -61,17 +64,18 @@ get_library()
         ("failed to find shared library of rmw implementation. Searched " + env_var).c_str());
       return nullptr;
     }
+
     try {
-      lib = new Poco::SharedLibrary(library_path);
-    } catch (Poco::LibraryLoadException & e) {
+      lib = std::make_shared<rcpputils::SharedLibrary>(library_path.c_str());
+    } catch (const std::runtime_error & e) {
       RMW_SET_ERROR_MSG(
-        ("failed to load shared library of rmw implementation. Exception: " +
-        e.displayText()).c_str());
+        ("failed to load shared library of rmw implementation." + library_path +
```

Comment:
Nitpick: the message has a period between the static text and the library path and no space - looks weird to me.